### PR TITLE
test: fix output logs

### PR DESCRIPTION
**Pull request by cpanato**

### CODE REVIEW

File: tests/rspec/lib/metal_support.rb [hunk 1]
```diff
@@ -77,6 +77,7 @@ def self.start_matchbox
       wait_for_matchbox
       puts 'Starting QEMU/KVM nodes'
       system(env_variables_setup, 'sudo -E ./scripts/libvirt create')
+      system(env_variables_setup, 'virsh dumpxml node1')
```
Comment (squat) [inline tests/rspec/lib/metal_support.rb]: Why just node1?

Comment (cpanato) [reply tests/rspec/lib/metal_support.rb]: just for debugging purposes @squat

**Comment by cpanato** [comment]
retest this please

**Comment by cpanato** [comment]
retest this please

**Comment by cpanato** [comment]
ok to test

File: tests/rspec/lib/metal_support.rb [hunk 2]
```diff
@@ -125,35 +125,49 @@ def self.check_service(service)
...
+        files = [stdout, stderr]
+        until all_files_eof(files)
+          ready = IO.select(files)
+          next unless ready
+          readable = ready[0]
+          readable.each do |f|
+            fileno = f.fileno
+            begin
+              data = f.read_nonblock(1024)
+              output = ''
```
Comment (squat) [inline tests/rspec/lib/metal_support.rb]: output is not frozen, right? Why not <<?

**Comment by cpanato** [comment]
@mxinden can you please take another look?

File: tests/rspec/lib/metal_support.rb [hunk 3]
```diff
@@ -125,33 +125,24 @@ def self.check_service(service)
...
-        output += "End of journal of #{service} service on #{ip}"
-        output.close
-      end
-    rescue => e
-      puts "Cannot retrieve logs of service #{service} - failed to exec with: #{e}"
-    end
+    cmd = "journalctl --no-pager -u #{service}"
+    stdout, stderr, exit_status = Open3.capture3(cmd)
+    raise "Cannot retrieve logs of service #{service}" unless exit_status.success?
+    output = ''
```
Comment (mxinden) [inline tests/rspec/lib/metal_support.rb]: Why initialize output to empty string? Why not just do `output = "Journal of #{service} service (exitcode #{exit_status})"`

Comment (cpanato) [reply tests/rspec/lib/metal_support.rb]: done

**Pull request merged**